pkg/kv/kvserver/concurrency/testdata/concurrency_manager/poison_policy_wait

```diff
@@ -0,0 +1,60 @@
+#  This test sets up the following situation:
+#
+#     e     <- put  (waiting)
+#  b---f    <- scan (waiting)
+#   c       <- put  (sequenced, poisoned)
+#
+# The top and bottom request use PoisonPolicyWait,
```

Comment:
Let's add a third case where e is error and b---f is wait. b---f poisoning itself should cause e to bail.